### PR TITLE
Add io.github.daid.EmptyEpsilon

### PR DESCRIPTION
**Pull request by jstsmthrgk**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.
 EmptyEpsilon is a spaceship bridge simulator game, each player gets one of the consoles and together they control the ship.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] ~~I am an author/developer/upstream contributor of the project.~~ If not, I contacted upstream developers (@daid) about this submission. **Link:** https://github.com/daid/EmptyEpsilon/issues/1967#issuecomment-2397556125

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.daid.EmptyEpsilon.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '24.08'
+sdk: org.freedesktop.Sdk
+command: EmptyEpsilon-wrapper
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --device=dri
+  - --device=input
```
Comment (hfiguiere) [inline io.github.daid.EmptyEpsilon.yml:11]: device  input will not work. It is not part of any stable release of flatpak.

File: io.github.daid.EmptyEpsilon.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      - -DCPACK_PACKAGE_VERSION_PATCH=09
+      - -DSERIOUS_PROTON_DIR=/run/build/EmptyEpsilon/SeriousProton/
+      - -Dglm_DIR=/run/build/EmptyEpsilon/glm/cmake/glm/
+      - -DFETCHCONTENT_SOURCE_DIR_BASIS=/run/build/EmptyEpsilon/basis-universal/
+      - -DFETCHCONTENT_SOURCE_DIR_MESHOPTIMIZER=/run/build/EmptyEpsilon/meshoptimizer/
+    post-install:
+        # install wrapper to use xdg base directory for config (for minimal permissions)
+      - install -Dpm755 -t "${FLATPAK_DEST}/bin" /run/build/EmptyEpsilon/EmptyEpsilon-wrapper
+        # put wrapper into desktop file
+      - sed -i 's/Exec=EmptyEpsilon/Exec=EmptyEpsilon-wrapper/g' /app/share/applications/io.github.daid.EmptyEpsilon.desktop
```
Comment (hfiguiere) [inline io.github.daid.EmptyEpsilon.yml:32]: don't use sed. Use `desktop-file-edit` instead.

File: io.github.daid.EmptyEpsilon.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --device=dri
+  - --device=input
+  - --share=ipc
+  - --share=network
+modules:
+  - name: glm
+    buildsystem: cmake-ninja
```
Comment (hfiguiere) [inline io.github.daid.EmptyEpsilon.yml:16]: there is not source for that module

```suggestion
```

File: io.github.daid.EmptyEpsilon.yml (hunk 1 above)
Comment (jstsmthrgk) [reply io.github.daid.EmptyEpsilon.yml]: ah, I didn't catch that, then I'll need --device=all for joystick input

**Comment by jstsmthrgk** [comment]
bot, build io.github.daid.EmptyEpsilon

File: io.github.daid.EmptyEpsilon.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+sdk: org.freedesktop.Sdk
+command: EmptyEpsilon-wrapper
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --device=dri
+  - --device=all # should be --device=input at some point in the future
+  - --share=ipc
+  - --share=network
```
Comment (bbhtt) [inline io.github.daid.EmptyEpsilon.yml:13]: why is network needed ?

Comment (jstsmthrgk) [reply io.github.daid.EmptyEpsilon.yml:13]: networked multiplayer

**Comment by jstsmthrgk** [comment]
I believe this PR might have been overlooked regarding the awaiting-changes tag...

**Comment by bbhtt** [comment]
Looks like it...

bot, build io.github.daid.EmptyEpsilon

**Pull request closed without merging**